### PR TITLE
feat: add collection item files.

### PR DESCRIPTION
**Pull request by gdethier**
logion-network/logion-internal#519

### CODE REVIEW

File: packages/node-api/src/LogionLoc.ts [hunk 1]
```diff
@@ -343,7 +356,15 @@ export function addCollectionItem(parameters: AddCollectionItemParameters): Subm
...
 
-    return api.tx.logionLoc.addCollectionItem(collectionId.toHexString(), itemId, itemDescription);
+    const files: ({ name?: any; contentType?: any; size_?: any; hash_?: any })[] = itemFiles.map(itemFile => ({
+        name: stringToHex(itemFile.name),
+        contentType: stringToHex(itemFile.contentType),
+        size_: itemFile.size,
+        hash_: itemFile.hash,
+    }));
+
+    return api.tx.logionLoc.addCollectionItem(collectionId.toHexString(), itemId, stringToHex(itemDescription), files);
```
Comment (benoitdevos) [inline packages/node-api/src/LogionLoc.ts:369]: Seems weird to me that we now convert description to Hex - Not backward compatible, and neither equivalent change in `getCollectionItem`. Did I miss something ?

**Review by benoitdevos — COMMENTED** [review]
Good job !
I'll approve but would appreciate an answer to my second question before.

File: packages/node-api/src/LogionLoc.ts [hunk 2]
```diff
@@ -343,7 +356,15 @@ export function addCollectionItem(parameters: AddCollectionItemParameters): Subm
         collectionId,
         itemId,
         itemDescription,
+        itemFiles,
     } = parameters;
 
-    return api.tx.logionLoc.addCollectionItem(collectionId.toHexString(), itemId, itemDescription);
+    const files: ({ name?: any; contentType?: any; size_?: any; hash_?: any })[] = itemFiles.map(itemFile => ({
```
[the remark this answers is not in the document]
Comment (gdethier) [reply packages/node-api/src/LogionLoc.ts:362]: Yes, types. If I use `PalletLogionLocCollectionItemFile` directly, I cannot benefit from automatic conversions anymore.

**Review by benoitdevos — APPROVED** [review]
:+1:

File: packages/node-api/src/LogionLoc.ts (hunk 1 above)
Comment (gdethier) [reply packages/node-api/src/LogionLoc.ts:369]: Actually, calling `stringToHex` on the string is just making explicit what happens under the hood. Indeed, the string must be converted somehow into a vector of bytes (which is the type expected by the runtime in the end). We use that approach on LOC metadata's name and value and it works as expected.

**Pull request merged**